pytential/symbolic/compiler.py

```diff
@@ -340,22 +362,28 @@ def gen_expr_arrow(expr, target_node):
 # {{{ code representation
 
 class Code:
+    inputs: set[str]
+    result: Expression | ObjectArray1D[Expression]
+
+    _schedule: Sequence[tuple[Statement, Collection[str]]]
+
     def __init__(
             self,
-            inputs: Set[str],
+            inputs: set[str],
             schedule: Sequence[tuple[Statement, Collection[str]]],
-            result: np.ndarray,
+            result: Expression | ObjectArray1D[Expression],
            ) -> None:
         self.inputs = inputs
-        self._schedule = schedule
         self.result = result
+        self._schedule = schedule
```

Comment:
I would like this to have a type parameter, with bound `Expression | ObjectArray1D[Expression]`, so that we can still tell whether it's a scalar or a vector output. This will help with typing precision of operator outputs.

Reply:
Fair. Though I'm happy to type for *intent* and fix the code if it doesn't actually do what the intended types suggest.